lib/core/stores/remote.ts

```diff
@@ -289,4 +400,31 @@ export class RemotePromiseStore implements IPromiseStore {
       throw new ResonateError(ErrorCodes.ENCODER, "Decode error", e);
     }
   }
+
+  private isDeletedResponse(obj: any): obj is { deleted: boolean } {
+    return obj !== undefined && obj.deleted === true;
+  }
+
+  // Function to check if the response matches the Schedule type
+  private isSchedule(obj: any): obj is Schedule {
+    // You may need to adjust this based on the actual structure of your Schedule type
+    return (
+      obj !== undefined &&
+      typeof obj.id === "string" &&
+      typeof obj.cron === "string" &&
+      typeof obj.promiseId === "string" &&
+      typeof obj.promiseTimeout === "number"
+    );
+  }
+
+  private isSearchSchedulesResp(obj: any): obj is { cursor: string; schedules: Schedule[] } {
```

Comment:
Can we move this to store.ts? I'm open to different locations for these store guards but let's make sure the promise+schedules ones are colocated